includes/wf_crm_webform_preprocess.inc

```diff
@@ -810,17 +813,18 @@ class wf_crm_webform_preprocess extends wf_crm_webform_base {
     $message = $this->replaceTokens($message, $contact);
     preg_match_all('#\{([^}]+)\}#', $message, $matches);
     if (!empty($matches[0])) {
-      $q = $_GET;
-      unset($q['q'], $q['cs'], $q['cid'], $q['cid1']);
-      if (empty($_GET['cid']) && empty($_GET['cid1'])) {
-        $q['cid1'] = 0;
+      $query = \Drupal::request()->query;
+      if (empty($query->get('cid')) && empty($query->get('cid1'))) {
+        $query->set('cid1', 0);
```

Comment:
Is this right? The original code was only changing the local `$q['cid1']` variable, but the new code is changing 'cid1' on the request object (or at least appears to be - perhaps the query is copied? Doesn't look like it, though).

Reply:
It also seems like it's missing the `unset()`'s too. If those are actually ever passed, this would lead to different URLs being generated.

Maybe it should be doing something like:

```
$q = \Drupal::request->query->all()
```

And then using `$q` same as it did in the old code?

Reply:
That works, and makes more sense. I'll revise.